File: pset6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv)==2:
        print("Plaintext:", end="")
        t=cs50.get_string()
        key=int(sys.argv[1])
        if key>26:
            while True:
                key=key-26
                if key<=26:
                    break
        elif key<0:
            while True:
                key=key+26
                if key>=0:
                    break
        
        encrypt(t,key)
        exit(0)
    else:
        print("Invalid Input")
        exit(1)

def encrypt(t,key):
    print("Ciphertext:", end="")
    for i in range(len(t)):
        z=t[i]
        #check if its lowercase and change all to uppercase
        if ord(z)>=ord('a') and ord(z)<=ord('z'):
            z=ord(z)-(ord('a')-ord('A'))
        elif ord(z)>=ord('A') and ord(z)<=ord('Z'):
            z=z
        else:
            exit(1)
        
        c=z+key
        if c>ord('Z'):
            c=c-26
        #if it was lowercase change it back to lowercase
        if t[i]>='a' and t[i]<='z':
            c=c+(ord('a')-ord('A'))
            print("{}".format(chr(c)),end="")
        else:
            print("{}".format(chr(c)),end="")
    print()
    
if __name__ == "__main__":
    main()
    
    

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[])

    
{   
    if (argc==2)
    {
        printf("plaintext:");
        string input=get_string();
        int k= atoi(argv[1]);
        if (k>26)
        {
            do
            {
                k-=26;
            }
            while (k>26);
        }
        else if (k<0)
        {
            do
            {
                k+=26;
            }
            while (k<0);
        }
        printf("ciphertext:");
        for (int i=0, len=strlen(input); i<len; i++)
        {
            char I=toupper(input[i]);
            if ((I>='A' && I<='Z'))
            {
                char j=(I+k);
                {
                    if (j>'Z')
                    {
                        j=j-26;
                    }
                    else if (j<'A')
                    {
                        j+=26;
                    }
                    else
                    {
                        j+=0;
                    }
                    if (input[i]>='a' && input[i] <='z')
                    {
                        printf("%c", tolower(j));
                    }    
                    else
                    {
                        printf("%c", j);
                    }
                }    
            }
            else
            {
                printf("%c", input[i]);
            }
        }
        printf("\n");
        return 0;
    }    
    else
    {
        printf("Invalid input\n");
        return 1;
    }
}
